### PR TITLE
Configure notification channels

### PR DESCRIPTION
**Pull request by abhibhaw**
- [x] Subscribed to Academic & Other Notification Channel

- [x] Saved Unique Device Tokens to Firestore

- [x] Added Unsubscribe to Optional Notification on Settings Page

### CODE REVIEW

**Comment by abhibhaw** [comment]
## Added Flex to fix this.

# Before
___________________________________________________________

Screenshot_1605124478

# After 
___________________________________________________________

Screenshot_1605124490

**Comment by abhibhaw** [comment]
# Settings Page UI

Screenshot_1605124472
Screenshot_1605124466

**Review by DesignrKnight — CHANGES_REQUESTED** [review]

**Comment by DesignrKnight** [comment]
@Thesmader 
Over to you

**Comment by DesignrKnight** [comment]
@abhibhaw 
Do mention when you do the changes discussed. Will approve and merge

**Comment by abhibhaw** [comment]
@DesignrKnight Changes are throwing a few errors 😢 I am on it!!

File: scp/lib/HomePage.dart [hunk 1]
```diff
@@ -278,6 +330,7 @@ class _HomePageState extends State<HomePage> {
     fetchUserData(context);
     reset();
     _fcm.subscribeToTopic('ics-not');
```
[the remark this answers is not in the document]
Comment (abhibhaw) [reply scp/lib/HomePage.dart]: Done!

File: scp/lib/HomePage.dart [hunk 2]
```diff
@@ -269,6 +284,43 @@ class _HomePageState extends State<HomePage> {
...
+      'devToken': token,
+      'displayName': username,
+      'roll': rollNo,
+      'mobile': phoneNo,
+      'optionalSub': true,
+      'createdAt': FieldValue.serverTimestamp(),
+    };
+    Firestore.instance
+        .collection('tokens')
+        .where('devToken', isEqualTo: token)
```
[the remark this answers is not in the document]
Comment (abhibhaw) [reply scp/lib/HomePage.dart]: Done

File: scp/lib/drawer_screens/settings.dart [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                            Firestore.instance
+                                .collection('tokens')
+                                .document(phoneNo)
+                                .updateData({'optionalSub': false});
+                          } else {
+                            Firestore.instance
+                                .collection('tokens')
+                                .document(phoneNo)
+                                .updateData({'optionalSub': true});
+                          }
```
[the remark this answers is not in the document]
Comment (abhibhaw) [reply scp/lib/drawer_screens/settings.dart]: Done

**Comment by abhibhaw** [comment]
@DesignrKnight @Thesmader All set to go 👍

**Review by Thesmader — CHANGES_REQUESTED** [review]
Several small issues.

**Review by Thesmader — APPROVED** [review]

**Comment by Thesmader** [comment]
Good to go!

**Review by DesignrKnight — APPROVED** [review]

**Comment by DesignrKnight** [comment]
@abhibhaw @nayakastha @SriramPatibanda 
I am merging this. Once that is done. This state of the project, planning to push to Play Store and inform SAC. Do check thoroughly. 

Also, now that we have two notification channel. We need Firebase Cloud Functions for the same. I will explain the existing one so that one of you can modify that

**Pull request merged**